front/components/WorkspacePicker.tsx

```diff
@@ -15,54 +13,22 @@ export default function WorkspacePicker({
   onWorkspaceUpdate: (w: WorkspaceType) => void;
 }) {
   return (
-    <div className="flex items-center">
-      <Menu as="div" className="relative inline-block text-left">
-        <div>
-          <Menu.Button
-            className={classNames(
-              "inline-flex items-center rounded-md py-1 text-sm font-medium",
-              workspace ? "px-0" : "border px-3",
-              "focus:outline-none focus:ring-0"
-            )}
-          >
-            {workspace ? (
-              <>
-                <div className="text-sm text-slate-800">{workspace.name}</div>
-                <ChevronDownIcon className="ml-2 mt-0.5 h-4 w-4 hover:text-gray-700" />
-              </>
-            ) : (
-              "Select workspace"
-            )}
-          </Menu.Button>
-        </div>
+    <DropdownMenu className="flex">
+      <DropdownMenu.Button
+        label={workspace ? workspace.name : "Select workspace"}
+      />
 
-        <Menu.Items
-          className={classNames(
-            "absolute left-1 z-10 mt-1 origin-top-left rounded-md bg-white shadow-sm ring-1 ring-black ring-opacity-5 focus:outline-none",
-            workspace ? "-left-8" : "left-1"
-          )}
-        >
-          <div className="py-1">
-            {user.workspaces.map((w) => {
-              return (
-                <Menu.Item key={w.sId}>
-                  {({ active }) => (
-                    <span
-                      className={classNames(
-                        active ? "text-gray-900" : "text-gray-700",
-                        "block cursor-pointer whitespace-nowrap px-4 py-2 text-sm"
-                      )}
-                      onClick={() => onWorkspaceUpdate(w)}
-                    >
-                      {w.name}
-                    </span>
-                  )}
-                </Menu.Item>
-              );
-            })}
-          </div>
-        </Menu.Items>
-      </Menu>
-    </div>
+      <DropdownMenu.Items>
+        {user.workspaces.map((w) => {
+          return (
+            <DropdownMenu.Item
+              key={w.sId}
+              onClick={() => onWorkspaceUpdate(w)}
```

Comment:
onWorkspaceUpdate is likely async but we don't await it we need to explicitely say that's fine:
```
() => {
  void onWorskpaceUpdate(w);
}
```

Reply:
Due to your onClick handler not being async itself I presume while the previous one was...

Reply:
assuming this is the source of error might be in the other file will comment there as well